src/graphapi/opgraph.cpp

```diff
@@ -349,32 +341,36 @@ void BackendOperationGraphDescriptor::setAttribute(miopenBackendAttributeName_t
             std::vector<OpNode*> nodes;
             nodes.reserve(elementCount);
 
-            std::for_each(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
-                          static_cast<miopenBackendDescriptor_t*>(arrayOfElements) + elementCount,
-                          [&descriptors, &nodes](miopenBackendDescriptor_t apiDescriptor) {
-                              BackendDescriptor& backendDescriptor = deref(apiDescriptor);
-                              if(backendDescriptor.isFinalized())
-                              {
-                                  descriptors.push_back(apiDescriptor);
-                                  nodes.push_back(backendDescriptor.getOperation());
-                              }
-                              else
-                              {
-                                  MIOPEN_THROW(miopenStatusBadParm);
-                              }
-                          });
+            std::for_each_n(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
+                            elementCount,
```

Comment:
- This partially reverts #2973
```suggestion
            // for_each_n is not available on RHEL/SLES, see issue #2973
            std::for_each(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
                          static_cast<miopenBackendDescriptor_t*>(arrayOfElements) + elementCount,
```